src/components/ebay-progress-bar-expressive/component.ts

```diff
@@ -0,0 +1,152 @@
+import { AttrString } from "marko/tags-html";
+import type { WithNormalizedProps } from "../../global";
+
+export const messageDurationStandard = 1500;
+export const messageDurationReducedMotionMultiplier = 1.5;
+export const messageFadeInDuration = 833;
+
+interface Message {
+    text: string;
+    duration?: number;
+}
+
+interface ProgressBarExpressiveInput
+    extends Omit<Marko.Input<"div">, `on${string}`> {
+    "a11y-text"?: AttrString;
+    messages?: Marko.RepeatableAttrTag<Message>;
+    size?: "medium" | "large";
+}
+
+export interface Input
+    extends WithNormalizedProps<ProgressBarExpressiveInput> {}
+
+interface State {
+    messages: Message[];
+    messageIsFadingIn: boolean;
+    currentMessageIndex: number;
+    nextMessageIndex: number;
+    isInitialMessage: boolean;
+    useReducedMotion: boolean;
+}
+
+class ProgressBarExpressive extends Marko.Component<Input, State> {
+    declare timeouts: { fadeIn?: NodeJS.Timeout; showMessage?: NodeJS.Timeout };
+    declare fadeInFirstMessage: boolean;
+
+    onCreate(input: Input) {
+        const messages = (input.messages as Message[]) || [];
+        const useReducedMotion =
+            typeof window !== "undefined" &&
+            window.matchMedia("(prefers-reduced-motion: reduce)").matches;
```

Comment:
I believe we might need to move this outside of the class and in the global definitions. (after imports)
I think this is due to the create might not be called on the client if its initialized on the server. 

If we move it in the root, we can access it anywhere in the file which is great!